src/main/java/by/onliner/utils/Recorder.java

```diff
@@ -0,0 +1,120 @@
+    }
+
+    /***
+     * Creates a file for recording the movie.
+     * @param fileFormat format of the file
+     * @return created file
+     */
+    @Override
+    protected File createMovieFile(Format fileFormat) throws IOException {
+
+        if (!movieFolder.exists()) {
+            movieFolder.mkdirs();
+        } else if (!movieFolder.isDirectory()) {
+            throw new IOException("\"" + movieFolder + "\" is not a directory.");
+        }
+        return new File(movieFolder,
+                Recorder.name);
+    }
+
+    /***
+     * Initialise recorder and start recording test
+     * @param name method name
+     */
+    public static void startRecording(String name) throws Exception {
+        File file = new File(savePath);
+
+        Dimension screenSize = Toolkit.getDefaultToolkit().getScreenSize();
+        int width = screenSize.width;
+        int height = screenSize.height;
+
+        Rectangle captureSize = new Rectangle(0, 0, width, height);
+
+        GraphicsConfiguration gc = GraphicsEnvironment.getLocalGraphicsEnvironment().
+                getDefaultScreenDevice()
+                .getDefaultConfiguration();
+
+        screenRecorder = new Recorder(gc, captureSize,
+                new Format(MediaTypeKey, MediaType.FILE, MimeTypeKey, MIME_AVI),
+                new Format(MediaTypeKey, MediaType.VIDEO, EncodingKey, ENCODING_AVI_TECHSMITH_SCREEN_CAPTURE,
+                        CompressorNameKey, ENCODING_AVI_TECHSMITH_SCREEN_CAPTURE, DepthKey, 24, FrameRateKey,
+                        Rational.valueOf(15), QualityKey, 1.0f, KeyFrameIntervalKey, 15 * 60),
+                new Format(MediaTypeKey, MediaType.VIDEO, EncodingKey, "black", FrameRateKey, Rational.valueOf(30)),
+                null, file, name);
+
+        screenRecorder.start();
+    }
+
+    /***
+     * Stoped recording test
+     * @param name method name
+     */
+    public static void stopRecording(String name) throws Exception {
+        screenRecorder.stop();
+        aviToMp4(name);
+    }
+
+    /***
+     * Converts avi to mp4 video format
+     * @param name file name
+     */
+    private static void aviToMp4(String name) {
+        File source = new File(savePath + Recorder.name);
+        File target = new File(savePath + name + ".mp4");
+        VideoAttributes video = new VideoAttributes();
+        video.setCodec("libx264");
+        video.setBitRate(3200000);
+        video.setFrameRate(15);
+        EncodingAttributes attrs = new EncodingAttributes();
+        attrs.setFormat("mp4");
+        attrs.setVideoAttributes(video);
+        Encoder encoder = new Encoder();
+        MultimediaObject multimediaObject = new MultimediaObject(source);
+        try {
+            encoder.encode(multimediaObject, target, attrs);
+        } catch (IllegalArgumentException | EncoderException e) {
+            e.printStackTrace();
```

Comment:
why have you decided to process the exception here like this?

Reply:
To optimise my code

Reply:
what does it optimize? what kind of optimization is that? what results have you achieved with this optimization?
How does it affect test execution? what will happen in case of any error in this block and how will it affect the execution?

Reply:
For catch a definite exceptions